libs/cli/src/main/java/org/elasticsearch/cli/Terminal.java

```diff
@@ -275,12 +280,18 @@ public boolean isHeadless() {
         return false;
     }
 
+    public Terminal asJson() {
```

Comment:
nit: maybe make this abstract so the UOE is clearly on ConsoleTerminal (and the message can clearly state ConsoleTerminal is what has the issue).